src/drivers/dw/dma.c

```diff
@@ -328,7 +331,8 @@ static int dw_dma_start(struct dma_chan_data *channel)
 	channel->core = cpu_get_id();
 
 	/* enable linear link position */
-	platform_dw_dma_llp_enable(dma, channel);
+	if (!platform_dw_dma_is_llp_enable(dma, channel))
+		platform_dw_dma_llp_enable(dma, channel);
```

Comment:
It's probably better if ```platform_dw_dma_llp_enable()``` does the "enabled" check rather than a new function (unless you plan to use the new is_enabled() in other places) ?

Reply:
sure